types/parameterized_struct_type.go

```diff
@@ -0,0 +1,80 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/substrait-io/substrait-go/types/parameter_types"
+)
+
+// ParameterizedStructType is a struct having at least one parameter of type ParameterizedAbstractType
+// example: Struct<Decimal(P,S), INT8>.
+// If All arguments are concrete they are represented by StructType
+type ParameterizedStructType struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	Type             []Type
+}
+
+func (*ParameterizedStructType) isRootRef() {}
+func (m *ParameterizedStructType) WithNullability(n Nullability) Type {
+	m.Nullability = n
+	return m
+}
+
+func (m *ParameterizedStructType) GetType() Type               { return m }
+func (m *ParameterizedStructType) GetNullability() Nullability { return m.Nullability }
+func (m *ParameterizedStructType) GetTypeVariationReference() uint32 {
+	return m.TypeVariationRef
+}
+func (m *ParameterizedStructType) Equals(rhs Type) bool {
+	if o, ok := rhs.(*ParameterizedStructType); ok {
+		if m.Nullability != o.Nullability || len(m.Type) != len(o.Type) ||
+			m.TypeVariationRef != o.TypeVariationRef {
+			return false
+		}
+		for i := range m.Type {
+			if !m.Type[i].Equals(o.Type[i]) {
+				return false
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func (m *ParameterizedStructType) ShortString() string {
+	t := StructType{}
+	return t.ShortString()
```

Comment:
Maybe use

```go
return (StructType{}).ShortString()
```

or 

```go
var t StructType
return t.ShortString()
```

Reply:
this method is no more needed in FuncDefArgType